pkg/clusteragent/clusterchecks/dispatcher_main.go

```diff
@@ -48,11 +54,16 @@ func (d *dispatcher) add(config integration.Config) {
 	if !config.ClusterCheck {
 		return // Ignore non cluster-check configs
 	}
-	log.Debugf("dispatching configuration %s:%s", config.Name, config.Digest())
 
-	// TODO: add dispatching logic
-	hostname, _ := util.GetHostname()
-	d.addConfig(config, hostname)
+	target := d.getLeastBusyNode()
+	if target == "" {
+		// If no node is found, store it in the danglingConfigs map for retrying later.
+		log.Infof("No available node to dispatch %s:%s on, will retry later", config.Name, config.Digest())
+	} else {
+		log.Infof("Dispatching configuration %s:%s to node %s", config.Name, config.Digest(), target)
```

Comment:
i'm on the fence between info and debug for this one. I guess info is fine but we may end up lowering to debug if this runs smoothly and it gets spammy.

Reply:
I went for consistency with [the collector's scheduler](https://github.com/DataDog/datadog-agent/blob/27e6bb2d015681f25850ed4376f41f57350d19da/pkg/collector/scheduler.go#L63), but indeed on high service churn that might become spammy.